Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/ThresholdedRandomCutForest.java

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.parkservices.threshold;
+
+import static com.amazon.randomcutforest.CommonUtils.checkArgument;
+
+import java.util.Arrays;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import com.amazon.randomcutforest.RandomCutForest;
+import com.amazon.randomcutforest.parkservices.AnomalyDescriptor;
+import com.amazon.randomcutforest.returntypes.DiVector;
+
+@Getter
+@Setter
+public class ThresholdedRandomCutForest {
+
+    public static int MINIMUM_OBSERVATIONS_FOR_EXPECTED = 100;
+
+    // a parameter that determines if the current potential anomaly is describing
+    // the same anomaly
+    // within the same shingle or across different time points
+    protected double ignoreSimilarFactor = 0.3;
+
+    // a different test for anomalies in the same shingle
+    protected double triggerFactor = 3.0;
+
+    // saved attribution of the last seen anomaly
+    protected long lastAnomalyTimeStamp;
+
+    // score of the last anomaly
+    protected double lastAnomalyScore;
+
+    // attribution of the last anomaly
+    protected DiVector lastAnomalyAttribution;
+
+    // last processed score
+    protected double lastScore;
+
+    // actual value of the last anomalous point (shingle)
+    double[] lastAnomalyPoint;
+
+    // if sufficient observations are present, what is the expected values
+    // (for the top numberOfAttributor fields) for the last anomaly
+    double[] lastExpectedPoint;
+
+    // indicates that previous point was a candidate anomaly
+    boolean previousIsPotentialAnomaly;
+
+    // indicates if we are in a region where scores are high; this can be useful in
+    // its own right
+    boolean inHighScoreRegion;
+
+    // flag that determines if we should dedup similar anomalies not in the same
+    // shingle, for example an
+    // anomaly, with the same pattern is repeated across more than a shingle
+    protected boolean ignoreSimilar;
+
+    // for anomaly description we would only look at these may top attributors
+    // note that expected value is not well-defined when this number is greater than
+    // 1
+    int numberOfAttributors = 2;
```

Comment:
The forecast algorithm does not assume only 1 value is missing -- but as we impute more values, since we are making those numbers up, there will naturally be more errors. There should be an upper bound (currently it is 2). Also displaying this information is not easy if this value is large. Comment updated.